core/src/main/java/com/linecorp/armeria/server/Server.java

```diff
@@ -655,52 +657,79 @@ private void doStop(CompletableFuture<Void> future,
                     lock.unlock();
                 }
 
-                // Close all accepted sockets.
-                ChannelUtil.close(connectionLimitingHandler.children()).handle((unused3, unused4) -> {
-                    // Shut down the worker group if necessary.
-                    final Future<?> workerShutdownFuture;
-                    if (config.shutdownWorkerGroupOnStop()) {
-                        workerShutdownFuture = config.workerGroup().shutdownGracefully();
-                    } else {
-                        workerShutdownFuture = ImmediateEventExecutor.INSTANCE.newSucceededFuture(null);
-                    }
-
-                    workerShutdownFuture.addListener(unused5 -> {
-                        final Set<EventLoopGroup> bossGroups =
-                                Server.this.serverChannels.stream()
-                                                          .map(ch -> ch.eventLoop().parent())
-                                                          .collect(toImmutableSet());
-
-                        // If started to shutdown before initializing a boss group,
-                        // complete the future immediately.
-                        if (bossGroups.isEmpty()) {
-                            finishDoStop(future);
-                            return;
+                shutdownServerHandlers().handle((unused3, unused4) -> {
```

Comment:
I understood now, that requests are closed with an exception once before closing the socket directly.

As a result, users will see more `503` responses instead of connection resets